test/unit/utils/api-request.spec.ts

```diff
@@ -623,56 +623,91 @@ describe('HttpClient', () => {
     }).should.eventually.be.rejectedWith(err).and.have.property('code', 'app/network-error');
   });
 
-  it('should timeout when the response is delayed', () => {
+  it('should timeout when the response is repeatedly delayed', () => {
     const respData = {foo: 'bar'};
     const scope = nock('https://' + mockHost)
       .get(mockPath)
-      .twice()
+      .times(5)
```

Comment:
I think it is better to test passing explicit RetryConfig instead so you don't have to do things like:
`(client as any).retry.backOffFactor = 0;`

This also makes it easier to add changes to default config in the future.

And then add a test for default config.

Reply:
Done